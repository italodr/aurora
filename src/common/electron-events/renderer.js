import names from "./names";
import { ipcRenderer } from "electron";

/**
 * Listeners
 */
function onChangeScreen(then) {
  return ipcRenderer.on(names.CHANGE_SCREEN, then);
}

export default {
  onChangeScreen
};
