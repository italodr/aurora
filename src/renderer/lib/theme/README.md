# aurora-theme

The main CSS themes: colors, fonts, spacing, ect.
This package makes use of [styled component's themeing.](https://www.styled-components.com/docs/advanced#theming)