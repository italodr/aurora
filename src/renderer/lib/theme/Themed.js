import base from "./base.js";
import React from "react";
import PropTypes from "prop-types";
import styled, { ThemeProvider, injectGlobal } from "styled-components";

// Note: Eslint kinda messes up on this particular usage of template literal functions
// So we put a disable comment here.
/* eslint-disable no-unused-expressions */
injectGlobal`
    html, body, #root {
        font-family: ${base.font};
        height: 100%;
        width: 100%;
        padding: 0;
        margin: 0;
        background: ${base.colors.background};
        color: ${base.colors.text};
    }

    * {
      box-sizing: border-box;
    }
`;

const Expand = styled.div`
  height: 100%;
  width: 100%;
  padding: 0;
  margin: 0;
`;

// A helper for wrapping something in the default theme.
// Replaces a regular ThemeProvider from styled-components
// https://www.styled-components.com/docs/advanced#theming
const Themed = ({ children, theme }) => {
  return (
    <ThemeProvider theme={theme}>
      <Expand>{children}</Expand>
    </ThemeProvider>
  );
};

Themed.propTypes = {
  children: PropTypes.any.isRequired,
  theme: PropTypes.object
};

// By default use the base theme
Themed.defaultProps = {
  theme: base
};

export default Themed;
