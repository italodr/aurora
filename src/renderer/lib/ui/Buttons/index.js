import styled from "styled-components";

const Button = styled.button`
  border: 0;
  border-radius: ${props => props.theme.borderRadius};
  display: inline-block;
  text-transform: capitalize;
  line-height: 16px;
  font-size: 14px;
  font-weight: 400;
  text-decoration: none;
  text-align: center;
  white-space: nowrap;
  box-sizing: border-box;
  cursor: pointer;
  padding: 10px 20px;
  transition: 0.2s;
  transition-property: background-color, color, box-shadow, opacity,
    -webkit-box-shadow;

  &:hover,
  &:focus,
  &:active {
    outline: 0;
  }
`;

const PrimaryButton = Button.extend`
  background: ${props => props.theme.colors.darkPrimary};
  color: white;

  &:hover,
  &:focus,
  &:active {
    background: ${props => props.theme.colors.primary};
  }
`;

const SuccessButton = Button.extend`
  background: ${props => props.theme.colors.darkSuccess};
  color: white;

  &:hover,
  &:focus,
  &:active {
    background: ${props => props.theme.colors.success};
  }
`;

const DangerButton = Button.extend`
  background: ${props => props.theme.colors.darkDanger};
  color: white;

  &:hover,
  &:focus,
  &:active {
    background: ${props => props.theme.colors.danger};
  }
`;

export { Button, PrimaryButton, SuccessButton, DangerButton };
