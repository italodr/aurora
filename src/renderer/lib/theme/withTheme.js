import Themed from "./Themed";
import React from "react";

const withTheme = Component => {
  return props => {
    return (
      <Themed>
        <Component {...props} />
      </Themed>
    );
  };
};

export default withTheme;
