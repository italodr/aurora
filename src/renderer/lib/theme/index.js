import base from "./base.js";
import Themed from "./Themed.js";
import withTheme from "./withTheme";

export { Themed, base, withTheme };
