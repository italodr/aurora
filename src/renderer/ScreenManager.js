import React from "react";
import Core from "./lib/core";
import { Store } from "./lib/mutate";
import { connect } from "react-redux";
import rendererEvents from "../common/electron-events/renderer";
import { setScreen } from "./redux/actions";
import { screen } from "./redux/selectors";

class ScreenManager extends React.Component {
  constructor(props) {
    super(props);

    // Listen for electron screen change events and pass them through as a Redux action
    // The general flow here is basically:
    // 1. User clicks on menu item
    // 2. Menu item launches an electron event to change the screen
    // 3. We receive the event here
    // 4. We send Redux a dispatch that tells it to change the screen
    // 5. ScreenManager treats the action like any other state change
    rendererEvents.onChangeScreen((event, screen) => {
      props.dispatch(setScreen(screen));
    });
  }

  render() {
    return this.props.screen;
  }
}

const Screens = {
  main: <Core />,
  store: <Store />
};

const mapStateToProps = state => {
  return {
    screen: Screens[screen(state)] || Screens.main
  };
};

export default connect(mapStateToProps)(ScreenManager);
