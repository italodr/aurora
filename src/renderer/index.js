import React from "react";
import ReactDOM from "react-dom";
import ScreenManager from "./ScreenManager";
import { Provider } from "react-redux";
import { setPreferences } from "./redux/actions";
import { loadPreferences } from "./lib/io";
import configureStore from "./redux/configureStore";
import Draggable from "./lib/ui/Draggable";

const store = configureStore();

class Aurora extends React.Component {
  constructor(props) {
    super(props);

    loadPreferences().then(prefs => {
      store.dispatch(setPreferences(prefs));
    });
  }

  render() {
    return (
      <Provider store={store}>
        <Draggable dragRegionSize="50px">
          <ScreenManager />
        </Draggable>
      </Provider>
    );
  }
}

// Add Electron to the page
document.body.innerHTML = `<div id="root"></div>`;
ReactDOM.render(<Aurora />, document.getElementById("root"));
