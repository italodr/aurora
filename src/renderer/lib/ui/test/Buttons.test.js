import { Button, PrimaryButton, SuccessButton, DangerButton } from "../Buttons";
import renderer from "react-test-renderer";
import { withTheme } from "../../theme";
import React from "react";

const expectRendersCorrectly = Component => {
  const tree = renderer.create(<Component> {"foo"}</Component>).toJSON();
  expect(tree).toMatchSnapshot();
};

test("Button renders correctly", () => {
  expectRendersCorrectly(withTheme(Button));
});

test("PrimaryButton renders correctly", () => {
  expectRendersCorrectly(withTheme(PrimaryButton));
});

test("SuccessButton renders correctly", () => {
  expectRendersCorrectly(withTheme(SuccessButton));
});

test("DangerButton renders correctly", () => {
  expectRendersCorrectly(withTheme(DangerButton));
});
