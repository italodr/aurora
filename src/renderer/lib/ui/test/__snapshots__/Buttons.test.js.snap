// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<div
  className="sc-ifAKCX ksmhtO"
>
  <button
    className="sc-bdVaJa ewSStz"
  >
     
    foo
  </button>
</div>
`;

exports[`DangerButton renders correctly 1`] = `
<div
  className="sc-ifAKCX ksmhtO"
>
  <button
    className="sc-bxivhb kBqUkW"
  >
     
    foo
  </button>
</div>
`;

exports[`PrimaryButton renders correctly 1`] = `
<div
  className="sc-ifAKCX ksmhtO"
>
  <button
    className="sc-bwzfXH IxvRo"
  >
     
    foo
  </button>
</div>
`;

exports[`SuccessButton renders correctly 1`] = `
<div
  className="sc-ifAKCX ksmhtO"
>
  <button
    className="sc-htpNat godGnZ"
  >
     
    foo
  </button>
</div>
`;
